waspc/src/Analyzer/Parser/Lexer.x

```diff
@@ -0,0 +1,113 @@
+{
+{-# LANGUAGE NamedFieldPuns #-}
+
+module Analyzer.Parser.Lexer
+  ( lexer
+  ) where
+
+import Analyzer.Parser.Util (AlexInput, Parser, ParseState (..), updatePosn, putInput)
+import Analyzer.Parser.Syntax (ParseError (..), Token (..), TokenClass (..))
+import Control.Monad.Trans.State.Lazy (get)
+import Control.Monad.Trans.Except (throwE)
+import Control.Monad.Trans.Class (lift)
+import Data.Word (Word8)
+import Codec.Binary.UTF8.String (encodeChar)
+}
+
+$digit = 0-9
+$alpha = [a-zA-Z]
+$identstart = [_$alpha]
+$ident = [_$alpha$digit]
+$any = [.$white]
+
+@string = \" (. # \")* \"
+@double = "-"? $digit+ "." $digit+
+@integer = "-"? $digit+
+@ident = $identstart+ $ident* "'"*
+
+tokens :-
+
+  $white+ ;
+
+-- Quoters — For now a dirty method that hardcodes json and psl in lexer
+-- TODO: figure out how to make this better
```

Comment:
Hmmm I see what you mean, that is certainly an issue. That is because it is greedy.

Let's add a failing test for this to ParserTest, we will get it working once we fix this? TDD ;).

Alex start codes -> I thought those are only for specifying which lexer rules are active at the moment? I am guessing you would need some kind of lookahead here instead?

EDIT: oh but wait, I am now looking at tests and I see you do have a test for that. And it is passing! How is that? So it works now?